xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -1800,33 +2186,16 @@ int CAMLCodec::Decode(uint8_t *pData, size_t iSize, double dts, double pts)
       m_1st_pts = am_private->am_pkt.lastpts;
   }
 
-  // if we have still frames, demux size will be small
-  // and we need to pre-buffer more.
-  double target_timesize = 1.0;
-  if (iSize < 20)
-    target_timesize = 2.0;
-
-  // keep hw buffered demux above 1 second
-  if (GetTimeSize() < target_timesize && m_speed == DVD_PLAYSPEED_NORMAL)
-    return VC_BUFFER;
-
-  // wait until we get a new frame or 25ms,
-  if (m_old_pictcnt == m_cur_pictcnt)
-    m_ready_event.WaitMSec(25);
+  if (!DequeueFrame(m_lastFrame))
```

Comment:
this might be an problem on startup, the hw codec has a long latency, so it will take several demux packets injected into hw decoder buffer before you even see it coming out the V4L layer. Not sure when EAGAIN happens.